### DISCUSSION TITLE
Windows Media Player

### DISCUSSION THREAD

**Discussion by damodaranr** [bluenviron/mediamtx#802] [Questions]
The server is great and easy to get going. However, for my specific use case I need Windows Media Player to consume the rtsp stream which as per the documentation is definitely supported.
I used ffmpeg to stream to the rtsp-simple-server over tcp.
However, I am only able to consume the stream only from VLC. If I try the same URL as in the documentation page using either mms://127.0.0.1:8554/mystream or rtsp://127.0.0.1:8554/mystream through Windows Media Player, the rtsp-simple-server logs the following error in the console:
2022/01/30 09:23:15 INF [RTSP] [conn [::1]:51019] opened
2022/01/30 09:23:15 INF [RTSP] [conn [::1]:51019] closed (invalid URL (/mystream))

I have a second question about udp-multicast: (This is not clear from the documentation page). Please provide additional information.
1) ffmpeg writes to rtsp-simple-server at http://localhost:8554/mystream
2) How to configure rtsp-simple-server to write this stream to a specific multicast IP:port or even ask it to write to the a multicast address within the range configured in the yml file?
3) How to configure Windows media player to receive from this multicast stream? will the **?vlcmulticast** flag work while connecting  to the rtsp server and negotiate to then use the multicast address?
 3.1) With windows media server, we could directly configure the multicast address on the client.

Does rtsp simple server support asf containers?
Thanks!

**Comment by aler9** [comment]
Hello,

> If I try the same URL as in the documentation page using either mms://127.0.0.1:8554/mystream or rtsp://127.0.0.1:8554/mystream through Windows Media Player, the rtsp-simple-server logs the following error in the console:

If you think this is a bug, open an issue and provide necessary data.

> How to configure rtsp-simple-server to write this stream to a specific multicast IP:port or even ask it to write to the a multicast address within the range configured in the yml file?

The server takes the first available IP address from the IP range written in the configuration file. Therefore, if you set a IP range that contains a single IP (like 224.1.0.0/32), then the server will use that IP. A client can't set the multicast IP directly since the protocol doesn't allow it.

> How to configure Windows media player to receive from this multicast stream? 

If Windows media player supports RTSP with multicast, it will use it. You can force multicast by using `protocols: [multicast]` in the configuration or either by using `?vlcmulticast`. Windows media player is seldom used, i don't know how good it is in this field.

> With windows media server, we could directly configure the multicast address on the client.

Even if you are able to receive the stream by manually listening on an IP / port, you won't be able to decode it since you need to know what codec the stream uses, what are the codec parameters, what are the tracks, and these infos are only sent during the RTSP handshake.

> Does rtsp simple server support asf containers?

no

[a comment by an automated account is not shown]
